src/reporting.js

```diff
@@ -215,39 +216,52 @@ async function _generateReportFromDownload(req, domain, mediaId, matrixFile, opt
     console.info(`Downloading ${httpUrl}, writing to ${filePath}`);
 
     let downloadHeaders;
-    let response;
 
     try {
         downloadHeaders = await new Promise((resolve, reject) => {
-            let responseHeaders;
-	    let connect = {url: httpUrl, encoding: null, qs: thumbnailQueryParams};
-	    const config = getConfig();
-	    const proxy = config.proxy;
-	    if (proxy != null) {
-		connect.proxy = proxy;
-	    }
-	    if (config.requestHeader != null) {
-		connect.headers = generateRequestHeaders(config.requestHeader, req);
-		console.info(`Request headers are`, connect.headers);
-	    }
-
-            request
-                .get(connect)
-                .on('error', reject)
-                .on('response', (response) => {
-                    responseHeaders = response.headers;
-                })
-                .on('end', () => {
-                    resolve(responseHeaders);
-                })
-                .pipe(fileWriteStream);
+            // Base options for the request
+            const opts = {
+                url: httpUrl,
+                headers: {
+                    'content-type': 'application/octet-stream',
+                }
+            };
+
+            const config = getConfig();
+
+            // Add additional request headers if configured
+            if (config.requestHeader != null) {
+                opts.headers += generateRequestHeaders(config.requestHeader, req);
```

Comment:
This looks like it's using `+` on two objects which I think does something very silly in js - you probably want `Object.assign()`.